dashboard/lib/projects_list.rb

```diff
@@ -42,5 +84,35 @@ def get_project_row_data(student, project, channel_id)
         updatedAt: project_value['updatedAt'],
       }.with_indifferent_access
     end
+
+    def fetch_published_projects_all_types(limit)
+      PUBLISHED_PROJECT_TYPES.map do |type|
+        [type, fetch_published_projects_by_type(type, limit)]
+      end.to_h
```

Comment:
Hmm. On further reflection, since you want `limit`-many of each type, I'm not sure how to do this in a readable fashion. Keep as is, for now, if the plan is to cache the results?